src/data/real/ereal.lean

```diff
@@ -514,3 +526,35 @@ lemma to_real_mul : ∀ {x y : ereal}, to_real (x * y) = to_real x * to_real y
 | ⊥ ⊥ := by simp
 
 end ereal
+namespace tactic
+
+open positivity
+
+meta instance : has_to_string strictness :=
+⟨λ s, match s with
+  | positive p := "strictness.positive (" ++ to_string p ++ ")"
+  | nonnegative p := "strictness.nonnegative (" ++ to_string p ++ ")"
+  end⟩
+
+meta instance : has_to_format strictness := ⟨λ s, to_string s⟩
+
+private lemma ereal_coe_nonneg {r : ℝ} : 0 ≤ r → 0 ≤ (r : ereal) := ereal.coe_nonneg.2
+private lemma ereal_coe_pos {r : ℝ} : 0 < r → 0 < (r : ereal) := ereal.coe_pos.2
+
+/-- Extension for the `positivity` tactic: cast from `ℝ` to `ereal`. -/
+@[positivity]
+meta def positivity_coe_ereal : expr → tactic strictness
+| `(@coe _ _ %%inst %%a) := do
+  match inst with
+  | `(@coe_to_lift _ _ %%inst) := do
+    strictness_a ← core a,
+    match inst, strictness_a with
+    | `(@coe_base _ _ ereal.has_coe), positive p := positive <$> mk_app ``ereal_coe_pos [p]
+    | `(@coe_base _ _ ereal.has_coe), nonnegative p := nonnegative <$> mk_app ``ereal_coe_nonneg [p]
```

Comment:
I must have done something wrong because with this I get
````
example {r : ℝ} (hr : 0 < r) : 0 < (r : ereal) := by positivity -- works
example {r : ℝ} (hr : 0 ≤ r) : 0 ≤ (r : ereal) := by positivity -- fails
```

Reply:
I had a look using `set_option trace.app_builder true`, it seems that the problem is
```quote
[app_builder] failed to create an '_private.2580574099.ereal_coe_nonneg'-application, failed to solve unification constraint for #1 argument (?x_2 ∈ ↑?x_0 =?= 0 ≤ r)
```
It seems to be fixed by being more explicit:
```suggestion
    | `(@coe_base _ _ ereal.has_coe), positive p := positive <$> mk_app ``ereal_coe_pos [p]
    | `(@coe_base _ _ ereal.has_coe), nonnegative p := nonnegative <$> mk_mapp ``ereal_coe_nonneg [a, p]
```

Reply:
Indeed!

Reply:
Can you please add your two examples as tests?